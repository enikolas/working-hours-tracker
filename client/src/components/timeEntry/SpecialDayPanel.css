.special-day-panel {
  padding: 90px 0;
  text-align: center;
  background: rgba(255,255,255,0.2);
  color: #fff;
  font-size: 20px;
  margin-top: 40px;
}
