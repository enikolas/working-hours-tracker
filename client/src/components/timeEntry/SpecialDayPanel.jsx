import React from 'react';
import PropTypes from 'prop-types';

import strings from '../../utils/strings';

import './SpecialDayPanel.css';

const renderTag = content => <p className="special-day-panel">{content}</p>;

const SpecialDayPanel = ({ entry }) => {
  if (entry.isVacation) {
    return renderTag(strings.vacation || '');
  }
  if (entry.isHoliday) {
    return renderTag(entry.holiday || '');
  }
  return null;
};

SpecialDayPanel.propTypes = {
  isBirthdayLeave: PropTypes.bool,
  isMedicalAbsence: PropTypes.bool,
  onSelect: PropTypes.func
};

SpecialDayPanel.defaultProps = {
  isBirthdayLeave: false,
  isMedicalAbsence: false,
  onSelect: () => {}
};

export default SpecialDayPanel;
