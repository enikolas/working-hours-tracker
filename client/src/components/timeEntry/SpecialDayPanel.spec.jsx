import React from 'react';
import { shallow } from 'enzyme';

import SpecialDayPanel from './SpecialDayPanel';

const defaultProps = {
	isBirthdayLeave: false,
	isMedicalAbstence: false,
	entry: {
		isVacation: false,
		isHoliday: false
	}
};

describe('SpecialDayPanel', () => {
	describe('render', () => {
		it('should not call renderTag if not vacation nor holiday', () => {
			const wrapper = shallow(<SpecialDayPanel
				entry={defaultProps.entry}
			/>);
			expect(wrapper).toMatchSnapshot();
		});
		it('should render holiday message', () => {
			const holidayPanel = { ...defaultProps.entry, isHoliday: true };
			const vacationPanel = { ...defaultProps.entry, isVacation: true };
			const wrapperHoliday = shallow(<SpecialDayPanel entry={holidayPanel} />);
			const wrapperVacation = shallow(<SpecialDayPanel entry={vacationPanel} />);

			expect(wrapperHoliday).toMatchSnapshot();
			expect(wrapperVacation).toMatchSnapshot();
		});
	});
});
