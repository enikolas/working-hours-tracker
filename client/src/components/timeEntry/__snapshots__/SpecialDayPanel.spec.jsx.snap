// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SpecialDayPanel render should not call renderTag if not vacation nor holiday 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <SpecialDayPanel
    entry={
      Object {
        "isHoliday": false,
        "isVacation": false,
      }
    }
    isBirthdayLeave={false}
    isMedicalAbsence={false}
    onSelect={[Function]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): null,
  Symbol(enzyme.__nodes__): Array [
    null,
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`SpecialDayPanel render should render holiday message 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <SpecialDayPanel
    entry={
      Object {
        "isHoliday": true,
        "isVacation": false,
      }
    }
    isBirthdayLeave={false}
    isMedicalAbsence={false}
    onSelect={[Function]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": "",
      "className": "special-day-panel",
    },
    "ref": null,
    "rendered": "",
    "type": "p",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": "",
        "className": "special-day-panel",
      },
      "ref": null,
      "rendered": "",
      "type": "p",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`SpecialDayPanel render should render holiday message 2`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <SpecialDayPanel
    entry={
      Object {
        "isHoliday": false,
        "isVacation": true,
      }
    }
    isBirthdayLeave={false}
    isMedicalAbsence={false}
    onSelect={[Function]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": "Férias",
      "className": "special-day-panel",
    },
    "ref": null,
    "rendered": "Férias",
    "type": "p",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": "Férias",
        "className": "special-day-panel",
      },
      "ref": null,
      "rendered": "Férias",
      "type": "p",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
